src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1956 @@
+            raise ValueError(
+                "config.text_config is expected to be of type ClvpEncoderConfig but is of type"
+                f" {type(config.text_config)}."
+            )
+
+        if not isinstance(config.speech_config, ClvpEncoderConfig):
+            raise ValueError(
+                "config.speech_config is expected to be of type ClvpEncoderConfig but is of type"
+                f" {type(config.speech_config)}."
+            )
+
+        if not isinstance(config.decoder_config, ClvpDecoderConfig):
+            raise ValueError(
+                "config.decoder_config is expected to be of type ClvpDecoderConfig but is of type"
+                f" {type(config.decoder_config)}."
+            )
+
+        self.conditioning_encoder = ClvpConditioningEncoder(config)
+
+        self.speech_decoder_model = ClvpForCausalLM(config)
+
+        self.text_encoder_model = ClvpEncoder(config.text_config)
+        self.speech_encoder_model = ClvpEncoder(config.speech_config)
+
+        self.logit_scale = nn.Parameter(torch.tensor(self.config.logit_scale_init_value))
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_conditioning_encoder_input_embeddings(self):
+        return self.conditioning_encoder.text_token_embedding
+
+    def set_conditioning_encoder_input_embeddings(self, new_embeddings):
+        self.conditioning_encoder.text_token_embedding = new_embeddings
+
+    def get_text_encoder_input_embeddings(self):
+        return self.text_encoder_model.token_embedding
+
+    def set_text_encoder_input_embeddings(self, new_embeddings):
+        self.text_encoder_model.token_embedding = new_embeddings
+
+    def get_speech_encoder_input_embeddings(self):
+        return self.speech_encoder_model.token_embedding
+
+    def set_speech_encoder_input_embeddings(self, new_embeddings):
+        self.speech_encoder_model.token_embedding = new_embeddings
+
+    # taken from the original repo,
+    # link : https://github.com/neonbjb/tortoise-tts/blob/4003544b6ff4b68c09856e04d3eff9da26d023c2/tortoise/api.py#L117
+    def fix_speech_decoder_output(self, autoreg_output: torch.LongTensor) -> torch.LongTensor:
+        """
+        This method modifies the output of the decoder model, such as replacing the `eos_token_id` and changing the
+        last few tokens of each sequence.
+
+        Args:
+            autoreg_output (`torch.LongTensor`):
+                This refers to the output of the decoder model.
+        """
+        decoder_fixing_codes = self.config.decoder_config.decoder_fixing_codes
+        autoreg_output = autoreg_output[:, 1:]
+
+        stop_token_indices = torch.where(autoreg_output == self.speech_decoder_model.config.eos_token_id, 1, 0)
+        autoreg_output = torch.masked_fill(autoreg_output, mask=stop_token_indices.bool(), value=83)
+
+        for i, each_seq_stop_token_indice in enumerate(stop_token_indices):
```

Comment:
Singular of indices is index! (`each_seq_stop_token_indice` -> `each_seq_stop_token_index`)

Reply:
:sweat_smile: